appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -169,24 +169,18 @@ internal class ExperienceRenderer(
 
     suspend fun start(owner: StateMachineOwning, context: RenderContext) {
         // If there's already a frame for the context, reset it back to its unregistered state.
-        val existingOwner = stateMachines.getOwner(context) as? AppcuesFrameStateMachineOwner
-        if (existingOwner != null) {
-            existingOwner.frame.get()?.reset()
-            existingOwner.stateMachine = null
-        }
+        stateMachines.getOwner(context)?.reset()
 
         // If the machine being started is already registered for a different context,
         // reset it back to its unregistered state before potentially showing new content.
-        if (owner.stateMachine != null) {
-            (owner as? AppcuesFrameStateMachineOwner)?.let {
-                it.frame.get()?.reset()
-                it.stateMachine = null
-            }
-        }
+        owner.reset()
 
+        // do we need to unhook any existing listeners to owner.stateMachine, if not null, before
+        // re-assigning a new one here?
```

Comment:
open question here, I'm not sure how to stop the observers on the state flow in the state machines at this point. It seems like we'd need to create and hold references to a coroutine somewhere and be able to cancel them? There does not appear to be any other way to stop a state flow collector. https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-state-flow/

Reply:
so we either change lifecycleTracker to be an instance per state machine, or we store job references by render context inside of lifecycleTracker, and then call a method like "lifecycleTracker.stop(renderContext)" that would cancel those jobs

Reply:
i like the idea of each state machine owning responsibility for its own tracker - will explore that